basyx/aas/adapter/http.py

```diff
@@ -218,6 +218,22 @@ def is_stripped_request(request: Request) -> bool:
 
 T = TypeVar("T")
 
+ENCODING = "utf-8"
+
+
+def base64url_decode(data: str) -> str:
+    try:
+        # If the requester omits the base64 padding, an exception will be raised.
+        # However, Python doesn't complain about too much padding,
+        # thus we simply always append two padding characters (==).
+        # See also: https://stackoverflow.com/a/49459036/4780052
+        decoded = base64.urlsafe_b64decode(data + "==").decode(ENCODING)
+    except binascii.Error:
+        raise BadRequest(f"Encoded data {str(data)} is invalid base64url!")
```

Comment:
If the type of `data` is changed to `str`, the wrapping `str()` around `data` can also be removed here.